src/BSON/UTCDateTime.c

```diff
@@ -223,11 +224,23 @@ static PHP_METHOD(MongoDB_BSON_UTCDateTime, toDateTime)
 	object_init_ex(return_value, php_date_get_date_ce());
 	datetime_obj = Z_PHPDATE_P(return_value);
 
-	sec_len = spprintf(&sec, 0, "@%" PRId64, intern->milliseconds / 1000);
-	php_date_initialize(datetime_obj, sec, sec_len, NULL, NULL, 0);
-	efree(sec);
+	sec  = intern->milliseconds / 1000;
+	usec = (llabs(intern->milliseconds) % 1000) * 1000;
```

Comment:
Do you need to include `<inttypes.h>` for this (see: [cppreference](https://en.cppreference.com/w/c/numeric/math/abs)). I imagine it may be included already through some sequence of libmongoc/php includes.

Reply:
Good point. It worked without, but I've added an explicit include so we don't have to rely on other headers including this for us.